### ISSUE TITLE
(v14): false-positive "Invalid regular expression: Invalid character class"

### ISSUE THREAD

**Issue by merlinnot** [sindresorhus/eslint-plugin-unicorn#448]
Hi! I noticed a bug in the latest release with some regular expressions, as this one:

```typescript
/^test-(?<name>[a-zA-Z-\d]+)$/u
```

It makes ESLint crash (not report an error, actually crash).

**Comment by sindresorhus** [comment]
// @jdanil
